TriangleApp/app/src/main/java/triangle/triangleapp/helpers/MediaHelper.java

```diff
@@ -0,0 +1,192 @@
+        if (firstInit) {
+            mCamera = CameraHelper.getCameraInstance();
+            mMediaRecorder = new MediaRecorder();
+
+            // Step 1: Unlock and set camera to MediaRecorder
+            mCamera.unlock();
+            mMediaRecorder.setCamera(mCamera);
+        }
+
+        // Step 2: Set sources
+        mMediaRecorder.setAudioSource(MediaRecorder.AudioSource.CAMCORDER);
+        mMediaRecorder.setVideoSource(MediaRecorder.VideoSource.CAMERA);
+
+        // Step 3: Set a CamcorderProfile (requires API Level 8 or higher)
+        mMediaRecorder.setProfile(CamcorderProfile.get(CamcorderProfile.QUALITY_480P));
+
+        // Step 4: Set output file
+        final String fileName = getOutputMediaFile(MEDIA_TYPE_VIDEO).toString();
+        mMediaRecorder.setOutputFile(fileName);
+
+        // Step 5: Set the preview output
+        mMediaRecorder.setPreviewDisplay(mCameraPreview.getHolder().getSurface());
+
+        mMediaRecorder.setOnInfoListener(new MediaRecorder.OnInfoListener() {
+            @Override
+            public void onInfo(MediaRecorder mr, int what, int extra) {
+                // Handle the on duration exceeded
+                if (what == MediaRecorder.MEDIA_RECORDER_INFO_MAX_DURATION_REACHED) {
+                    stopStreaming(false);
+                    startStreaming(true);
+
+                    if (webSocket.isConnected()) {
+                        File file = new File(fileName);
+
+                        int size = (int) file.length();
+                        byte bytes[] = new byte[size];
+                        byte tmpBuff[] = new byte[size];
+                        try {
+                            FileInputStream fis = new FileInputStream(file);
+
+                            int read = fis.read(bytes, 0, size);
+                            if (read < size) {
+                                int remain = size - read;
+                                while (remain > 0) {
+                                    read = fis.read(tmpBuff, 0, remain);
+                                    System.arraycopy(tmpBuff, 0, bytes, size - remain, read);
+                                    remain -= read;
+                                }
+                            }
+                        } catch (IOException e) {
+                            Log.e(TAG, "IoExc", e);
+                        }
+                        webSocket.sendStream(bytes);
+                        file.delete();
+                    }
+                }
+            }
+        });
+
+        mMediaRecorder.setOnErrorListener(new MediaRecorder.OnErrorListener() {
+            @Override
+            public void onError(MediaRecorder mediaRecorder, int what, int extra) {
+                Log.d(TAG, "Error in mediarecorder, " + what + ", " + extra);
+            }
+        });
+
+        mMediaRecorder.setMaxDuration(mediaRecorderMaxDuration);
+    }
+
+    private File getOutputMediaFile(int type) {
```

Comment:
Method is replaced to new class: MediaFileHelper. Nullpointer is valid, because the file is not exists in the directory.